x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleShardTaskParams.java

```diff
@@ -34,9 +35,12 @@ public record DownsampleShardTaskParams(
     ShardId shardId,
     String[] metrics,
     String[] labels,
-    String[] dimensions
+    String[] dimensions,
+    Map<String, String> alternativeSources
```

Comment:
Let's rename this to something with multi field in the name? Or `subFieldSources` since that is used elsewhere?